### ISSUE TITLE
Support agent node in a workflow

### ISSUE THREAD

**Issue by yc3231996** [langgenius/dify#7470]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

in current workflow design, we can add LLM or tools as nodes, but can't configure LLM node to call tools just like agents do.

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by hllshiro** [comment]
I need this too. The main purpose is to divide the process in two parts: Q&A guidance and result generation. Their prompts are the same, but the latter only outputs json_object

**Comment by hllshiro** [comment]
It seems that it can be implemented using node `HTTP Request`. The API interface will be provided after the agent is released.

**Comment by yc3231996** [comment]
it is more like a multi-agent requirement. I.E, in a workflow, an LLM node with ability of function calling is basically an agent.
It is the same thing to support choosing an agent as a node in a workflow. 
I would be helpful to have multi agents work together, even simply in a sequential mode.

**Comment by yc3231996** [comment]
> It seems that it can be implemented using node `HTTP Request`. The API interface will be provided after the agent is released.

do you mean call another agent via HTTP request in workflow? it is smart,  would be good if have this as a built-in feature.

**Comment by hllshiro** [comment]
> do you mean call another agent via HTTP request in workflow? it is smart, would be good if have this as a built-in feature.

Yes, since dify doesn't provide an agent node at the moment,  can only use this method to finish my job. Of course, I also agree that would be good if have this as a built-in feature

**Comment by kenneth-bro** [comment]
+1

**Comment by alterxyz** [comment]
Sounds like "Agent as Tool", sounds good.

**Comment by softgoto** [comment]
+1

**Comment by wltj920** [comment]
+1

**Comment by wltj920** [comment]
+1, We need "ReAct agent tool" that other agents can calling it to do something. Any agent can called by any agents.

**Comment by sapipoZZZ** [comment]
+1

**Comment by GuoNingNing** [comment]
+1

**Comment by aijak69** [comment]
.

**Comment by aijak69** [comment]
> @crazywoola Are there any plans to add this feature? Can't use Agent in workflow by http request, agent http Api not support blocking mode

You can use code block to parse the output for now

**Comment by infinitnet** [comment]
+1, just wanted to open a feature request for this as it seems logical that it should be possible to use agents in workflows. This way agents would be like an LLM + tool combination repository and workflows could be used to put these parts together, which is especially useful if someone re-uses the same sets of LLM + tool combinations (ie. "agents") within various workflows.

**Comment by neotize** [comment]
+1，Additionally, I hope that the Agent nodes can support the 'Add Message' feature (customizable multi-turn USER and ASSISTANT dialogues), to achieve a few-shot effect.

**Comment by MengXiangyu0624** [comment]
+1

**Comment by chou852ishare** [comment]
+1

**Comment by bejay2vwo** [comment]
+ 1

**Comment by dadastory** [comment]
I really need it, it's so import for a knowledge chat workflow app

**Comment by OraCatQAQ** [comment]
+1

**Comment by cloover** [comment]
+1

**Comment by LNZ001** [comment]
这是来自QQ邮箱的自动回复邮件。已收到~

**Comment by hoomjac** [comment]
+1

**Comment by lins403** [comment]
+1

**Comment by shuxinhuang** [comment]
+1

**Comment by kinglisky** [comment]
+1

**Comment by jinxianwei** [comment]
+1

**Comment by eightHundreds** [comment]
+1

**Comment by LNZ001** [comment]
这是来自QQ邮箱的自动回复邮件。已收到~

**Comment by LNZ001** [comment]
这是来自QQ邮箱的自动回复邮件。已收到~

**Comment by LNZ001** [comment]
这是来自QQ邮箱的自动回复邮件。已收到~

**Comment by lonergd** [comment]
+10086
